### PR TITLE
BUGFIX: Detect checked state for bound object collections

### LINKED ISSUE #1229: Using collection property with CheckboxViewHelper never checks checked automatically
### Description

The checkbox VH can be bound to an object property. If that property contains a collection of objects, the automatic setting of the `checked` attribute based on the `value` attribute does not work.

### Steps to Reproduce

1. Assign a `User` object to a form, it has a `groups` property.
2. Assign an array of `Group` objects to the form.
3. Render a list of checkboxes like
    ```
    <f:for each="{groups}" as="group">
        <f:form.checkbox property="groups" value="{group}"/>`
    </f:for>
    ```

#### Expected behavior

For any group the user is assigned, the `checked` attribute should be checked.

#### Actual behavior

No checkbox is checked.

### Affected Versions

Flow: 4.0 and up, maybe older versions

### PR DESCRIPTION
**Pull request by kdambekalns**
When a collection property with objects is bound to a checkbox VH,
this change makes sure to check object identifiers against the passed
value when automatically determining the `checked` state.

Fixes #1229

### CODE REVIEW

**Comment by bwaidelich** [comment]
as a bugfix shouldn't this target a lower branch?

**Comment by kdambekalns** [comment]
Absolutely, just didn't select the correct branch…

**Comment by kdambekalns** [comment]
Ah, there is Travis…

**Comment by albe** [comment]
Looks good, but yeah, a test would be cool

File: Neos.FluidAdaptor/Classes/ViewHelpers/Form/CheckboxViewHelper.php [hunk 1]
```diff
@@ -89,7 +93,22 @@ public function render($checked = null, $multiple = null)
         }
         if (is_array($propertyValue)) {
             if ($checked === null) {
-                $checked = in_array($valueAttribute, $propertyValue, true);
+                if (TypeHandling::isSimpleType(TypeHandling::getTypeForValue(current($propertyValue))) === false || Arrays::containsMultipleTypes($propertyValue)) {
```
Comment (bwaidelich) [inline Neos.FluidAdaptor/Classes/ViewHelpers/Form/CheckboxViewHelper.php]: I think we can just change this to always walk through the array. `Arrays::containsMultipleTypes()` iterates through it anyways.

**Comment by bwaidelich** [comment]
@kdambekalns sorry, I messed with Git..

**Comment by kdambekalns** [comment]
Regarding 3.3… well, I was just too lazy, AFAIR

**Comment by bwaidelich** [comment]
I think 4.0 as target is fine!

**Pull request merged**